src/test/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceApiServiceSpec.scala

```diff
@@ -722,41 +745,130 @@ class WorkspaceApiServiceSpec extends FlatSpec with WorkspaceApiService with Ent
       addMockOpenAmCookie ~>
       sealRoute(submitJobRoute) ~>
-      check { assertResult(StatusCodes.Created) {status} }
+      check {
+        assertResult(StatusCodes.Created) {
+          status
+        }
+      }
+  }
+
+  val z1 = Entity("z1", "samples", Map("foo" -> AttributeString("x"), "bar" -> AttributeNumber(3), "splat" -> attributeList), WorkspaceName(wsns, wsname))
+  val workspace2 = Workspace(
+    wsns + "2",
+    wsname + "2",
+    DateTime.now().withMillis(0),
+    "test",
+    Map.empty
+  )
+
+  it should "return 201 for copying entities into a workspace with no conflicts" in {
+    Post("/workspaces", HttpEntity(ContentTypes.`application/json`, workspace2.toJson.toString())) ~>
+      addMockOpenAmCookie ~>
+      sealRoute(postWorkspaceRoute) ~>
+      check {
+        assertResult(StatusCodes.Created) {
+          status
+        }
+        assertResult(workspace2) {
+          MockWorkspaceDAO.store((workspace2.namespace, workspace2.name))
+        }
+
+        Post(s"/workspaces/${workspace2.namespace}/${workspace2.name}/entities", HttpEntity(ContentTypes.`application/json`, z1.toJson.toString())) ~>
+          addMockOpenAmCookie ~>
+          sealRoute(createEntityRoute) ~>
+          check {
+            assertResult(StatusCodes.Created) {
+              status
+            }
+            assertResult(z1) {
+              MockEntityDAO.store(workspace2.namespace, workspace2.name)(z1.entityType, z1.name)
+            }
+
+            val destWorkspace = WorkspaceName(wsns, wsname)
+            val sourceWorkspace = WorkspaceName(workspace2.namespace, workspace2.name)
+            val entityCopyDefinition = EntityCopyDefinition(sourceWorkspace, destWorkspace, "samples", Seq("z1"))
+            Post("/entities/copy", HttpEntity(ContentTypes.`application/json`, entityCopyDefinition.toJson.toString())) ~>
+              addMockOpenAmCookie ~>
+              sealRoute(copyEntitiesRoute) ~>
+              check {
+                assertResult(StatusCodes.Created) {
+                  status
+                }
+                assertResult(z1) {
+                  MockEntityDAO.store(workspace.namespace, workspace.name)(z1.entityType, z1.name)
+                }
+              }
+          }
+      }
+  }
+
+  //TODO: verify that this ACTUALLY works. should return and assert StatusCodes.Conflict once the MockDAOs are changed
+  it should "return 409 for copying entities into a workspace with conflicts" in {
+    val destWorkspace = WorkspaceName(wsns, wsname)
+    val sourceWorkspace = WorkspaceName(workspace2.namespace, workspace2.name)
+    val entityCopyDefinition = EntityCopyDefinition(sourceWorkspace, destWorkspace, "samples", Seq("z1"))
+    Post("/entities/copy", HttpEntity(ContentTypes.`application/json`, entityCopyDefinition.toJson.toString())) ~>
+      addMockOpenAmCookie ~>
+      sealRoute(copyEntitiesRoute) ~>
+      check {
+        assertResult(StatusCodes.Created) {
+          status
+        }
+      }
```

Comment:
this test is actually faked at the moment because it was forced to use the mock. the other tests verified that this *should* work but there's no way to verify using this test case at the moment. Will be changed to assertResult(StatusCodes.Conflict) once new testing methods are in place